affinity/cpp-20/d0796r1.md

```diff
@@ -136,25 +142,60 @@ There are some additional challenges which we have been investigating but are no
 
     /* Execution context */
 
-    struct execution_context {
+    class execution_context {
+     public:
+
+      using executor_type = see-below;
 
-      using executor_type = __unspecfied__;
+      using pmr_memory_resource_type = see-below;
 
-      template <typename ExecutionResource>
-      execution_context(ExecutionResource &&execResource);
+      using allocator_type = see-below;
+
+      execution_context(const execution_resource &) noexcept;
 
       ~execution_context();
 
       const execution_resource &resource() const noexcept;
 
-      executor_type executor() noexcept;
+      executor_type executor() const;
+
+      pmr_memory_resource_type &memory_resource() const;
```

Comment:
why a reference?

Reply:
This has to be a reference, because if it were returned by value then if `memory_resource()` were to be called from two different places (potentially different threads) each of the calls would create a copy of the memory_resource, which means they would not be allocating memory in the same resource.